src/tracing/traced_value.cc

```diff
@@ -0,0 +1,191 @@
+// Copyright 2016 the V8 project authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "tracing/traced_value.h"
+
+#include <math.h>
+#include <sstream>
+#include <stdio.h>
+#include <string>
+
+#if defined(_STLP_VENDOR_CSTD)
+// STLPort doesn't import fpclassify into the std namespace.
+#define FPCLASSIFY_NAMESPACE
+#else
+#define FPCLASSIFY_NAMESPACE std
+#endif
+
+namespace node {
+namespace tracing {
+
+namespace {
+
+void EscapeAndAppendString(const char* value, std::string* result) {
```

Comment:
nit: It would be nicer to return a `std::string` instead of making `result` an in/out parameter, tbh…

Reply:
I agree. I adopted this directly from the v8 impl. If it's ok to diverge then I can change